nixos/lib/testing/nodes.nix

```diff
@@ -3,27 +3,34 @@ testModuleArgs@{ config, lib, hostPkgs, nodes, ... }:
 let
   inherit (lib) mkOption mkForce optional types mapAttrs mkDefault mdDoc;
 
+  inherit (import ../.. {}) evalSystemConfiguration baseModules;
+
   system = hostPkgs.stdenv.hostPlatform.system;
 
+  testModules = [
+    ./nixos-test-base.nix
+    { key = "nodes"; _module.args.nodes = config.nodesCompat; }
+    ({ config, ... }:
+      {
+        virtualisation.qemu.package = testModuleArgs.config.qemu.package;
+
+        # Ensure we do not use aliases. Ideally this is only set
+        # when the test framework is used by Nixpkgs NixOS tests.
+        nixpkgs.config.allowAliases = false;
+      })
+    testModuleArgs.config.extraBaseModules
+  ];
+
   baseOS =
-    import ../eval-config.nix {
-      inherit system;
+    evalSystemConfiguration {
       inherit (config.node) specialArgs;
-      modules = [ config.defaults ];
-      baseModules = (import ../../modules/module-list.nix) ++
-        [
-          ./nixos-test-base.nix
-          { key = "nodes"; _module.args.nodes = config.nodesCompat; }
-          ({ config, ... }:
-            {
-              virtualisation.qemu.package = testModuleArgs.config.qemu.package;
-
-              # Ensure we do not use aliases. Ideally this is only set
-              # when the test framework is used by Nixpkgs NixOS tests.
-              nixpkgs.config.allowAliases = false;
-            })
-          testModuleArgs.config.extraBaseModules
-        ];
+      modules = [
+        { nixpkgs = { inherit system; }; }
```

Comment:
I need help with critically reviewing each instance of `nixpkgs.system` as to whether it should be `nixpkgs.localSystem` or `nixpkgs.targetSystem`, as `nixpkgs.system` is being discouraged, we should do this right away, instead of making an additional refactoring step, setting foot on a discouraged island.

Reply:
Probably `hostPlatform` is all you need to set. `target` is irrelevant, because NixOS systems don't install a compiler, let alone a cross compiler. It's unfortunate that it's `nixos-rebuild --target`. Perhaps that should be changed to `--host`.

Reply:
We can worry about polishing that later.

Reply:
Yep, good idea!